src/flex-index.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef OSM2PGSQL_FLEX_INDEX_HPP
+#define OSM2PGSQL_FLEX_INDEX_HPP
+
+/**
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ * This file is part of osm2pgsql (https://osm2pgsql.org/).
+ *
+ * Copyright (C) 2006-2022 by the osm2pgsql developer community.
+ * For a full list of authors see the git log.
+ */
+
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+/**
+ * This class represents a database index.
+ */
+class flex_index_t
+{
+public:
+    explicit flex_index_t(std::string method) : m_method(std::move(method)) {}
+
+    std::string const &method() const noexcept { return m_method; }
+
+    std::string columns() const;
+
+    void set_columns(std::string const &columns)
```

Comment:
That's rather 'add_column'?

Reply:
This is not intended for adding columns one at a time, but an overload of the `void set_columns(std::vector<std::string> const &columns)` function so that we don't have to create a vector if all we want is to add a single column.

Reply:
I added an assert.